crates/fuzzing/src/oracles.rs

```diff
@@ -146,12 +146,114 @@ pub fn instantiate(wasm: &[u8], known_valid: bool, config: &generators::Config,
     let module = match Module::new(store.engine(), wasm) {
         Ok(module) => module,
         Err(_) if !known_valid => return,
-        Err(e) => panic!("failed to compile module: {:?}", e),
+        Err(e) => {
+            if let generators::InstanceAllocationStrategy::Pooling { .. } =
+                &config.wasmtime.strategy
+            {
+                // When using the pooling allocator, accept failures to compile when arbitrary
+                // limits have been exceeded.
+                let string = e.to_string();
+                if string.contains("exceeds the limit of") {
```

Comment:
I think that we should be able to relatively easily control the limits on the wasm-smith-generated-module, could one of the methods on `Config` automatically apply the configuration of the pooling allocator to the wasm-smith module limits perhaps? 

(that'd avoid the need to duplicate this error handling below as well)

Reply:
So we can't completely avoid this check as currently there's no way to constrain table type minimums in `wasm-smith` like we can with memory types. The pooling instance allocator may place limits on the maximum elements in a table and will error if a table type has a minimum that exceeds the allocator's limit.

I have some changes that makes hitting this error much less likely where we constraint the generated module based on the arbitrary pooling limits (if arbitrarily configured to use pooling), but we'll need to update `wasm-smith` to fully remove these two checks, I believe.